cypress/e2e/copying.cy.js

```diff
@@ -1,51 +1,52 @@
-describe("Tests about copying", function () {
+describe('Tests about copying', () => {
   beforeEach(() => {
+    cy.initializeApp();
     cy.clearDebugStore();
   });
 
   afterEach(() => {
-    cy.get("[data-cy-nav-tab='testTab']").click();
-    cy.get("[data-cy-test='selectAll']").click();
-    cy.get("[data-cy-test='deleteSelected']").click();
-    cy.get("[data-cy-delete-modal='confirm']").click();
-    cy.get("[data-cy-nav-tab='debugTab']").click();
+    cy.get('[data-cy-nav-tab=\'testTab\']').click();
+    cy.get('[data-cy-test=\'selectAll\']').click();
+    cy.get('[data-cy-test=\'deleteSelected\']').click();
+    cy.get('[data-cy-delete-modal=\'confirm\']').click();
+    cy.get('[data-cy-nav-tab=\'debugTab\']').click();
   });
 
-  it("Copy report to test tab", () => {
-    cy.visit("");
-    cy.get("[data-cy-nav-tab='testTab']").click();
-    cy.get("#metadataTable tbody", { timeout: 10000 })
-      .find("tr")
-      .should("not.exist");
+  it('Copy report to test tab', () => {
+    cy.navigateToTestTabAndWait();
+    cy.get('#metadataTable tbody', { timeout: 10000 })
+      .find('tr')
+      .should('not.exist');
     cy.createReport();
-    cy.get("[data-cy-nav-tab='debugTab']").click();
-    cy.get("#metadataTable tbody", { timeout: 10000 })
-      .find("tr")
-      .should("not.exist");
-    cy.get("[data-cy-debug='refresh']").click();
-    cy.get("#metadataTable tbody").find("tr").should("have.length", 1);
-    cy.get("[data-cy-debug='selectAll']").click();
-    cy.get("[data-cy-debug='openSelected']").click();
-    cy.get("#debug-tree .jqx-tree-dropdown-root > li").should("have.length", 1);
-    cy.get("[data-cy-debug-editor='copy']").click();
-    cy.get("[data-cy-nav-tab='testTab']").click();
+    //Possibly bad test, it navigates to tab, asserts that the table is empty and then refreshes and asserts that the table is filled.
```

Comment:
Good thing that you note that its a bad test but how does it even work?
Isn't this very time sensitive or even a race condition?

Reply:
Yes you're right, i've added an intercept and await for this